src/Microsoft.AspNetCore.Mvc.Core/ContentResult.cs

```diff
@@ -69,60 +68,30 @@ public class ContentResult : ActionResult
 
             if (Content != null)
             {
+                response.ContentLength = resolvedContentTypeEncoding.GetByteCount(Content);
 
-                // Calculate the byte length to set the content length header
-                var charBuffer = ArrayPool<char>.Shared.Rent(DefaultCharBufferSize);
-                int sourceIndex = 0;
-                long byteCount = 0;
-                var encoder = resolvedContentTypeEncoding.GetEncoder();
-
-                while (sourceIndex < Content.Length)
-                {
-                    var countOfCharsToCopy = Math.Min((Content.Length - sourceIndex), charBuffer.Length);
-
-                    Content.CopyTo(sourceIndex, charBuffer, 0, countOfCharsToCopy);
-
-                    byteCount += encoder.GetByteCount(charBuffer, 0, countOfCharsToCopy, flush: false);
-
-                    sourceIndex += countOfCharsToCopy;
-                }
-
-                response.ContentLength = byteCount;
-
-                // write the content to stream
                 var byteBuffer = ArrayPool<byte>.Shared.Rent(DefaultByteBufferSize);
 
                 try
                 {
-                    sourceIndex = 0;
+                    var sourceIndex = 0;
+
                     while (sourceIndex < Content.Length)
                     {
-                        var numOfCharsToCopy = Math.Min((Content.Length - sourceIndex), charBuffer.Length);
-
-                        Content.CopyTo(
-                            sourceIndex,
-                            charBuffer,
-                            0,
-                            numOfCharsToCopy);
-
-                        sourceIndex += numOfCharsToCopy;
+                        var charCount = Math.Min(Content.Length - sourceIndex, DefaultByteBufferSize);
```

Comment:
I tried `resolvedContentTypeEncoding.GetMaxCharCount` but that resulted in exceptions in few cases (because the count of characters was greater than the buffer size)